### ISSUE TITLE
Plan for the FlexFlow release 2021.3

### ISSUE THREAD

**Issue by jiazhihao** [flexflow/FlexFlow#78]
CI Infrastructures:
- [x] Build tests and CircleCI configurations: Efrain
- [ ] Finish unitests for individual operators (potentially use the Python interface): Efrain, Wei
- [x] Performance unitests again existing tensor libraries (PyTorch): Efrain
- [x] Accuracy tests for vision models and language models: Vinay, Wei 
- [x] Multi-GPU performance tests (1-4GPUs): Nirmal

Runtime improvements:
- [x] Finishing the NCCL integration (nccl -> master): Zhihao
- [x] Scalability evaluation on large models: Zhihao
- [ ] Enhancing front-end support for vision models (torch.vision zoo) and language models (hugging face model zoo): Mandeep, Wei

Documentations, tutorials, websites:
- [x] API documentations: Wei
- [ ] Detailed tutorials to demonstrate auto-tuning, debugging, and profiling: Zhihao 

Data preprocessing:
- [ ] Distributed data loader in Python
- [ ] Distributed data generator support for Keras

Others:
- [ ] Spack package for FlexFlow: Pat

TODOs for future releases
- [ ] TensorBoard support using [TensorBoardX](https://github.com/lanpa/tensorboardX)
- [ ] Support for AMD CPUs/GPUs
- [ ] Integration of FlexFlow and [Legate](https://developer.nvidia.com/legate-early-access)
